ci-operator/step-registry/upi/install/libvirt/upi-install-libvirt-commands.sh

```diff
@@ -194,34 +267,74 @@ create_vm () {
   MAC_ADDRESS=${2}
   IGNITION_VOLUME=${LEASED_RESOURCE}-${3}-ignition-volume
 
-  echo "Creating ${NAME} vm..."
-  virt-install \
-    --connect ${LIBVIRT_CONNECTION} \
-    --name ${NAME} \
-    --memory ${DOMAIN_MEMORY} \
-    --vcpus ${DOMAIN_VCPUS} \
-    --network network=${CLUSTER_NAME},mac=${MAC_ADDRESS} \
-    --disk="vol=${POOL_NAME}/${NAME}-volume" \
-    --osinfo ${VIRT_INSTALL_OSINFO} \
-    --graphics=none \
-    --import \
-    --noautoconsole \
-    --disk vol=${POOL_NAME}/${IGNITION_VOLUME},format=raw,readonly=on,serial=ignition,startup_policy=optional
+  if [ "$INSTALLER_TYPE" == "agent" ]; then
+    # Calculate the last dynamic vars
+    DOMAIN_UUID="$(uuidgen)"
+    EXTRA_ARGS="rd.neednet=1 nameserver=192.168.$(leaseLookup 'subnet').1 ip=dhcp ignition.firstboot ignition.platform.id=metal"
+
+    # Boot artifacts names/paths were exported above upon creation
+    # Now create a VM based on the domain template
+    echo "Exporting the variables needed to fill out the lease-based domain template"
+    export DOMAIN_NAME="${NAME}"
+    export DOMAIN_UUID
+    export DOMAIN_VCPUS=${DOMAIN_VCPUS}
+    export DOMAIN_MEMORY=${DOMAIN_MEMORY}
+    export DOMAIN_MAC="${MAC_ADDRESS}"
+    export QCOW_PATH="${LIBVIRT_IMAGE_PATH}"
+    export QCOW_NAME="${DOMAIN_NAME}.qcow2"
+    export NETWORK_NAME="${CLUSTER_NAME}"
+    export EXTRA_ARGS
+    envsubst < ${CLUSTER_PROFILE_DIR}/domain-template.xml > ${SHARED_DIR}/"${NAME}.xml"
+
+    echo "Prepared libvirt domain xml:"
+    cat ${SHARED_DIR}/"${NAME}.xml"
+
+    # Create the volume used as the base for the image
+    ${VIRSH} vol-delete ${QCOW_NAME} --pool ${POOL_NAME} || true
+    ${VIRSH} vol-create-as \
+     --name ${QCOW_NAME} \
+     --pool ${POOL_NAME} \
+     --format qcow2 \
+     --capacity ${VOLUME_CAPACITY}
+
+    # Start the domain
+    ${VIRSH} define ${SHARED_DIR}/"${NAME}.xml"
+    ${VIRSH} start "${DOMAIN_NAME}"
+    ${VIRSH} autostart "${DOMAIN_NAME}"
+
+  else
+    # Pre-create the disk volume
+    clone_volume ${NAME}-volume
+
+    echo "Creating ${NAME} vm..."
+    virt-install \
+      --connect ${LIBVIRT_CONNECTION} \
+      --name ${NAME} \
+      --memory ${DOMAIN_MEMORY} \
+      --vcpus ${DOMAIN_VCPUS} \
+      --network network=${CLUSTER_NAME},mac=${MAC_ADDRESS} \
+      --disk="vol=${POOL_NAME}/${NAME}-volume" \
+      --osinfo ${VIRT_INSTALL_OSINFO} \
+      --graphics=none \
+      --import \
+      --noautoconsole \
+      --disk vol=${POOL_NAME}/${IGNITION_VOLUME},format=raw,readonly=on,serial=ignition,startup_policy=optional
+  fi
```

Comment:
Adding the above, including volume clone function calls, means the function is no longer simply "creating the vm" as its name implies... Maybe a more appropriate function name would be `create_node` to reflect its expanded functionality?